pkgs/cli/test/integration.zig

```diff
@@ -226,3 +305,153 @@ test "CLI beam command with mock network - complete integration test" {
 
     std.debug.print("SUCCESS: All integration test checks passed\n", .{});
 }
+
+test "SSE events integration test - wait for justification and finalization" {
+    const allocator = std.testing.allocator;
+
+    // Get executable path
+    const exe_path = try getZeamExecutable();
+
+    // Start node and wait for readiness
+    const cli_process = try spinBeamSimNode(allocator, exe_path);
+    defer cleanupProcess(allocator, cli_process);
+
+    // Wait for node to be fully active
+    waitForNodeStart();
+
+    // Create SSE client
+    var sse_client = try SSEClient.init(allocator);
+    defer sse_client.deinit();
+
+    // Connect to SSE endpoint
+    try sse_client.connect();
+
+    std.debug.print("INFO: Connected to SSE endpoint, waiting for events...\n", .{});
+
+    // Read events until both justification and finalization are seen, or timeout
+    const timeout_ms: u64 = 120000; // 120 seconds timeout
+    const start_ns = std.time.nanoTimestamp();
+    const deadline_ns = start_ns + timeout_ms * std.time.ns_per_ms;
+    var got_justification = false;
+    var got_finalization = false;
+
+    // streaming loop: accumulate SSE and parse JSON payloads
+    var read_buf: [4096]u8 = undefined;
+    var accum = std.ArrayList(u8).init(allocator);
+    defer accum.deinit();
+
+    while (std.time.nanoTimestamp() < deadline_ns and !(got_justification and got_finalization)) {
+        const bytes_read = sse_client.connection.read(&read_buf) catch |err| switch (err) {
```

Comment:
why are you not calling readEvents here, all this stuff needs to mvoe inside readEvents including event and data parsing